src/state/selectors.ts

```diff
@@ -241,18 +272,28 @@ export const selectIsTooltipActive = (
   state: RechartsRootState,
   tooltipEventType: TooltipEventType,
   trigger: TooltipTrigger,
+  defaultIndex: number | undefined,
 ): boolean => {
-  if (tooltipEventType === 'axis') {
+  const isTooltipActive = createSelector(selectActiveCoordinate, (coordinate: ChartCoordinate) => {
```

Comment:
I believe the createSelector call should be at the root level so that it is evaluated once, instead of on every run.

Now how to validate that belief. I suppose we should be able to write a unit test to demonstrate that? There is already a test that a selector is stable, that might show the difference?

Reply:
I need the values passed in initially but I'm not sure how to do that when calling createSelector at the root level since it only returns the results of the previous selectors and not any input you pass. I could return the input in the previous selector but that ruins the selectors purpose

Reply:
You can have what redux docs call "input selector", that is a function that ignores state and only returns the user input. That way you get the initial arguments in the combiner too. Check out how we're using it here: https://github.com/recharts/recharts/blob/3.x/src/state/dataSelectors.ts#L27

Alternatively, if you prefer, you can skip the `createSelector` part, and call selectors directly. `const isTooltipActive = selectActiveCoordinate(state, ...`. But beware that this way we lose the memoization that `createSelector` comes with.

Reply:
Updated to do the first way, thanks for the explanation